### ISSUE TITLE
Cannot serialize enumeration?

### ISSUE THREAD

**Issue by EricSalemi** [irmen/Pyro4#202]
Hi,

I recently bumped into an issue where pyro cannot serialize an enum instance.

The enum class is dead simple:

```python
from enum import Enum

class FlowControl(Enum):
    NONE = 0
    XON_XOFF = 1
    RTS_CTS = 2
    DSR_DTR = 3
```

I can reproduce the problem directly with serpent:

```python
>>> import serpent
>>> from septentrio.connection import FlowControl
>>> serpent.dumps(FlowControl.RTS_CTS)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Python27\lib\site-packages\serpent.py", line 91, in dumps
    return Serializer(indent, set_literals, module_in_classname).serialize(obj)
  File "C:\Python27\lib\site-packages\serpent.py", line 320, in serialize
    self._serialize(obj, out, 0)
  File "C:\Python27\lib\site-packages\serpent.py", line 361, in _serialize
    func(self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 627, in ser_default_class
    self._serialize(value, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 342, in _serialize
    return self.dispatch[t](self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 499, in ser_builtins_dict
    serialize(value, out, level + 1)
  File "C:\Python27\lib\site-packages\serpent.py", line 361, in _serialize
    func(self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 627, in ser_default_class
    self._serialize(value, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 342, in _serialize
    return self.dispatch[t](self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 499, in ser_builtins_dict
    serialize(value, out, level + 1)
  File "C:\Python27\lib\site-packages\serpent.py", line 361, in _serialize
    func(self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 627, in ser_default_class
    self._serialize(value, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 342, in _serialize
    return self.dispatch[t](self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 499, in ser_builtins_dict
    serialize(value, out, level + 1)
  File "C:\Python27\lib\site-packages\serpent.py", line 361, in _serialize
    func(self, obj, out, level)
  File "C:\Python27\lib\site-packages\serpent.py", line 599, in ser_default_class
    raise ValueError("Circular reference detected (class)")
ValueError: Circular reference detected (class)
```

I've tried this on Python 2.7.15 both x86 and amd64 platform.

Is that expected?

The documentation mentions that enumerations for Python 3.4 are supported and I assumed that the enum34 backported package would be supported too.

**Comment by irmen** [comment]
Yep, this is expected. The readme says for enums you'll need Python 3.4+  and the code is built around this restriction, currently.

The main reason why it doesn't work for "not the real enums" can be found here:  https://github.com/irmen/Serpent/blob/master/serpent.py#L164   the code only considers the "true" builtin (python 3.4+) enum type.   It will consider the enum34 backport as a custom class, tries to serialize it with its generic serialization code, but it appears that the implementation of the backported enum contains a reference cycle somehow to itself (which is not allowed).

As a workaround, you can register your own custom serpent serializer via its `register_class` function, and write the serialization code yourself for the enum34 backported class, in such a way that it avoids the reference cycle.

If it really bothers you please open a new issue at the Serpent project https://github.com/irmen/Serpent/ or better yet, create a PR if you're able :)

**Comment by irmen** [comment]
@EricSalemi decided to revisit the code in Serpent and it turned out to be quite simple to adapt it to work with the enum34 backport library as well.  Can you perhaps try it out?  I've not released a new version yet so you'll have to install serpent 1.26 from git directly to try this.

**Comment by EricSalemi** [comment]
I'll try it out when I come back from my holidays.

**Comment by irmen** [comment]
I've released an official 1.26 version of serpent to pypi now so you can simply install that and see how it goes. Closing the issue again for now.
